imblearn/over_sampling/_rose.py

```diff
@@ -0,0 +1,210 @@
+        Control the randomization of the algorithm.
+
+        - If int, ``random_state`` is the seed used by the random number
+          generator;
+        - If ``RandomState`` instance, random_state is the random number
+          generator;
+        - If ``None``, the random number generator is the ``RandomState``
+          instance used by ``np.random``.
+
+    n_jobs : int, default=None
+        Number of CPU cores used during the cross-validation loop.
+        ``None`` means 1 unless in a :obj:`joblib.parallel_backend` context.
+        ``-1`` means using all processors. See
+        `Glossary <https://scikit-learn.org/stable/glossary.html#term-n-jobs>`_
+        for more details.
+
+    See Also
+    --------
+    SMOTE : Over-sample using SMOTE.
+
+    SMOTENC : Over-sample using SMOTE for continuous and categorical features.
+
+    SVMSMOTE : Over-sample using SVM-SMOTE variant.
+
+    ADASYN : Over-sample using ADASYN.
+
+    KMeansSMOTE : Over-sample applying a clustering before to oversample using
+        SMOTE.
+
+    Notes
+    -----
+
+    References
+    ----------
+    .. [1] N. Lunardon, G. Menardi, N.Torelli, "ROSE: A Package for Binary
+       Imbalanced Learning," R Journal, 6(1), 2014.
+
+    .. [2] G Menardi, N. Torelli, "Training and assessing classification
+       rules with imbalanced data," Data Mining and Knowledge
+       Discovery, 28(1), pp.92-122, 2014.
+
+    Examples
+    --------
+
+    >>> from imblearn.over_sampling import ROSE
+    >>> from sklearn.datasets import make_classification
+    >>> from collections import Counter
+    >>> r = ROSE(shrink_factors={0:1, 1:0.5, 2:0.7})
+    >>> X, y = make_classification(n_classes=3, class_sep=2,
+    ... weights=[0.1, 0.7, 0.2], n_informative=3, n_redundant=1, flip_y=0,
+    ... n_features=20, n_clusters_per_class=1, n_samples=2000, random_state=10)
+    >>> print('Original dataset shape %s' % Counter(y))
+    Original dataset shape Counter({1: 1400, 2: 400, 0: 200})
+    >>> X_res, y_res = r.fit_resample(X, y)
+    >>> print('Resampled dataset shape %s' % Counter(y_res))
+    Resampled dataset shape Counter({2: 1400, 1: 1400, 0: 1400})
+    """
+
+    @_deprecate_positional_args
+    def __init__(self, *, sampling_strategy="auto", shrink_factors=None,
+                 random_state=None, n_jobs=None):
+        super().__init__(sampling_strategy=sampling_strategy)
+        self.random_state = random_state
+        self.shrink_factors = shrink_factors
+        self.n_jobs = n_jobs
+
+    def _make_samples(self,
+                      X,
+                      class_indices,
+                      n_class_samples,
+                      h_shrink):
+        """ A support function that returns artificial samples constructed
+        from a random subsample of the data, by adding a multiviariate
+        gaussian kernel and sampling from this distribution. An optional
+        shrink factor can be included, to compress/dilate the kernel.
```

Comment:
`_make_samples` is a private method, so public documentation is not generated for it.

Some of these notes may be better in the public `ROSE` docstring.

Reply:
moved most information to pertinent positions, I'll leave the doc here anyway for future maintenance, if it's ok for  you.